### PR TITLE
Host-Aware RSS Caching

### LINKED ISSUE #40: RSS Caching Can Return Different Hostnames
If a request comes in after the RSS cache has expired on an unexpected hostname, it can seed that hostname in the cache and return it with requests to the preferred hostname. This appears to only affect the RSS feed because it is returning fully qualified URLs.

**Example:**

```
camel on  master
➜ curl http://linode.liss.test/rss | grep linode
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1706    0  1706    0     0   171k      0 --:--:-- --:--:-- --:--:--  416k
]]></description><link>http://www.vt.edu/</link><guid isPermaLink="false">http://linode.liss.test/2015/2/6/sample-link-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 06 Feb 2015 22:00:00 GMT</pubDate></item><item><title><![CDATA[Test Post]]></title><description><![CDATA[<p>This is a <em>test post</em> entitled “Test Post”.</p>
]]></description><link>http://linode.liss.test/2014/5/1/sample-post</link><guid isPermaLink="false">http://linode.liss.test/2014/5/1/sample-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 02 May 2014 02:50:00 GMT</pubDate></item></channel></rss>

camel on  master
➜ curl http://www.liss.test/rss | grep linode
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1706    0  1706    0     0   161k      0 --:--:-- --:--:-- --:--:--  333k
]]></description><link>http://www.vt.edu/</link><guid isPermaLink="false">http://linode.liss.test/2015/2/6/sample-link-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 06 Feb 2015 22:00:00 GMT</pubDate></item><item><title><![CDATA[Test Post]]></title><description><![CDATA[<p>This is a <em>test post</em> entitled “Test Post”.</p>
]]></description><link>http://linode.liss.test/2014/5/1/sample-post</link><guid isPermaLink="false">http://linode.liss.test/2014/5/1/sample-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 02 May 2014 02:50:00 GMT</pubDate></item></channel></rss>
```

Screenshot 2023-02-22 at 09 15 41

This can result in feed readers picking up all feed items as new because the GUIDs change. (Origin: https://fosstodon.org/@drdrang/109905798081310673)

### PR DESCRIPTION
**Pull request by cliss**
Fixes #40.

### CODE REVIEW

**Comment by cliss** [comment]
@bryanculver, if you had a minute, I'd appreciate a once-over? No worries if not.

File: camel.js [hunk 1]
```diff
@@ -738,18 +750,22 @@ app.get('/rss', function (request, response) {
     }
     response.type('application/rss+xml');
 
-    if (typeof(renderedRss.date) === 'undefined' || new Date().getTime() - renderedRss.date.getTime() > 3600000) {
+    var previouslyRendered = renderedRss[request.hostname];
```
Comment (bryanculver) [inline camel.js:753]: At this level, it's almost certainly unlikely that `request` is undefined but you do have this nice safety check (https://github.com/cliss/camel/blob/master/camel.js#L186) that could be thrown in here too.

**Review by bryanculver — APPROVED** [review]
LGTM!

Validated locally too:

```
camel on  40-fix-rss-caching
➜ curl http://linode.liss.test/rss | grep linode
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1706    0  1706    0     0  37951      0 --:--:-- --:--:-- --:--:-- 43743
]]></description><link>http://www.vt.edu/</link><guid isPermaLink="false">http://linode.liss.test/2015/2/6/sample-link-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 06 Feb 2015 22:00:00 GMT</pubDate></item><item><title><![CDATA[Test Post]]></title><description><![CDATA[<p>This is a <em>test post</em> entitled “Test Post”.</p>
]]></description><link>http://linode.liss.test/2014/5/1/sample-post</link><guid isPermaLink="false">http://linode.liss.test/2014/5/1/sample-post</guid><dc:creator><![CDATA[Your Name]]></dc:creator><pubDate>Fri, 02 May 2014 02:50:00 GMT</pubDate></item></channel></rss>

camel on  40-fix-rss-caching
➜ curl http://www.liss.test/rss | grep linode
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1697    0  1697    0     0   137k      0 --:--:-- --:--:-- --:--:--  276k
```

File: camel.js (hunk 1 above)
Comment (cliss) [reply camel.js:753]: Good point. However:
* `request` is given to me from express; if it ain't there, some shit has really broken
* I don't think it's that big a deal for the RSS feeds anyway; if they break, they break

Good spot either way. :)

**Pull request merged**